beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/GenericMetadataHandler.java

```diff
@@ -0,0 +1,165 @@
+   * 3   db1 | tbl2    | path2 | year=2020,hour=2 <- partition for above 
+   * 
+   * Assume the above records are returned, the table entry 1 (unpartitioned) will be dropped. 
+   * For entry 2, tbl2 will not be dropped, since the number of records for db1.tbl2 = 2. 
+   * For entry 3, this partition will be dropped from the table. Now the repository looks like:
+   * 
+   * no: db  | tblname | path  | part vals
+   * 1   db1 | tbl2    | path2 | null             <- partitioned
+   * 
+   * and this time, when entry 1 is returned, the number of records for db1.tbl2 = 1, so this table is dropped. 
+   * 
+   */
+
+  /**
+   * Processes a pageable entityHouseKeepingPath page.
+   *
+   * @param pageable Pageable to iterate through for dryRun
+   * @param page Page to get content from
+   * @param dryRunEnabled Dry Run boolean flag
+   * @implNote This parent handler expects the child's cleanupPath call to update & remove the record from this call
+   *           such that subsequent DB queries will not return the record. Hence why we only call next during dryRuns
+   *           where no updates occur.
+   * @implNote Note that we only expect pageable.next to be called during a dry run.
+   * @return Pageable to pass to query. In the case of dry runs, this is the next page.
+   */
+  public Pageable processPage(Pageable pageable, Page<HousekeepingMetadata> page, boolean dryRunEnabled) {
+    List<HousekeepingMetadata> pageContent = page.getContent();
+
+    if (dryRunEnabled) {
+      pageContent.forEach(metadata -> cleanupMetadata(metadata, pageable));
+      return pageable.next();
+    } else {
+      pageContent.forEach(metadata -> cleanupContent(metadata, pageable));
+      return pageable;
+    }
+  }
+
+  private boolean cleanupMetadata(HousekeepingMetadata housekeepingMetadata, Pageable pageable) {
+    MetadataCleaner metadataCleaner = getMetadataCleaner();
+    PathCleaner pathCleaner = getPathCleaner();
+    int partitionCount = getPartitionCountForDatabaseAndTable(housekeepingMetadata.getDatabaseName(),
+        housekeepingMetadata.getTableName(), pageable);
+
+    String partitionName = housekeepingMetadata.getPartitionName();
+    if (partitionName == null && partitionCount == 0) {
+      cleanUpTable(housekeepingMetadata, metadataCleaner, pathCleaner);
+      return true;
+    }
+    if (partitionName != null) {
+      cleanupPartition(housekeepingMetadata, metadataCleaner, pathCleaner);
+      return true;
+    }
+    return false;
+  }
+
+  private void cleanUpTable(
+      HousekeepingMetadata housekeepingMetadata,
+      MetadataCleaner metadataCleaner,
+      PathCleaner pathCleaner) {
+    metadataCleaner.cleanupMetadata(housekeepingMetadata);
+    pathCleaner.cleanupPath(housekeepingMetadata);
+  }
+
+  private void cleanupPartition(
+      HousekeepingMetadata housekeepingMetadata,
+      MetadataCleaner metadataCleaner,
+      PathCleaner pathCleaner) {
+    metadataCleaner.cleanupPartition(housekeepingMetadata);
```

Comment:
```suggestion
    metadataCleaner.dropPartition(housekeepingMetadata);
```
? Unless there is some reason for using `cleanup` that I've missed?